authors/apps/comments/views.py

```diff
@@ -125,3 +128,49 @@ def get(self, request, id, *args, **kwargs):
         serializer = self.serializer_class(thread, many=True)
         return Response({"comments": serializer.data,
                          "thread count": thread.count()})
+
+
+class LikeCommentApiView(generics.RetrieveUpdateAPIView):
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = serializers.LikeCommentSerializer
+    renderer_classes = (LikeComementsJSONRenderer, )
+
+    def get_queryset(self):
+        return LikeComment.objects.all()
+
+    def get_object(self):
+        id = self.kwargs.get('id')
+        return get_object_or_404(Comment.objects.all(), id=id)
+
+    def retrieve(self, request, *args, **kwargs):
+        likes = self.get_queryset()
+        serializer = self.serializer_class(data=likes, many=True)
+        serializer.is_valid()
+        return Response(serializer.data)
+
+    def update(self, request, *args, **kwargs):
+        id = self.kwargs.get('id')
+        comment = self.get_object()
+        like = {
+            'comment': comment.id,
+            'user': request.user.pk
+        }
+
+        if self.get_queryset().filter(comment=id, user=request.user.pk):
```

Comment:
@Opio-Emmanuel-Omona You could do this to access the id.
```suggestion
        if self.get_queryset().filter(comment=comment.id, user=request.user.pk):
```

Reply:
@archbishop12  I am using what was suggested